### DISCUSSION TITLE
Missing visualizations of workflows and concepts?

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1184] [General]
I like to visualize complex workflows and concept.
So my question is:

Are there any topics that you think are still missing for a better understanding?

**Comment by flomickl** [comment]
First!  :face_in_clouds: 

I created a visualization of Data in StreamPipes

![ep_layers](https://user-images.githubusercontent.com/16739072/215619457-0846d514-3bb8-42cc-b03b-20cf5bc66cee.png)

But to understand the concept of ListProperties an example is missing

**Comment by flomickl** [comment]
The concept of **Time in StreamPipes** is missing

I am working on a first draft, but any ideas?
Especially if you want to merge two streams

**Reply by flomickl** [reply]
I made a first draw
![Event Processing and Time](https://user-images.githubusercontent.com/16739072/217089895-13489062-0745-4f80-95f5-456ed378ccc5.png)

**Comment by flomickl** [comment]
Concept of micro-batch processing, especially window functions

DO you think as well?
I am working on a first draft already

**Comment by flomickl** [comment]
I think a lot have changed here as well @dominikriemer @tenthe @bossenti  ?

![grafik](https://user-images.githubusercontent.com/16739072/216394925-2426eaeb-1c19-468c-a4e1-4220e4e950ff.png)

**Reply by tenthe** [reply]
Yes, +1 for updating the image

**Comment by flomickl** [comment]
according to https://github.com/apache/streampipes/discussions/1173#discussioncomment-4854002
need this graphic an update as well

![grafik](https://user-images.githubusercontent.com/16739072/216395652-7772582d-d85e-4ed2-a085-a46183c89242.png)

**Reply by flomickl** [reply]
What do you think about this?
![ep_layers](https://user-images.githubusercontent.com/16739072/216820998-aa9c00cc-33f0-4f92-8393-52fe9943d3b4.png)

together with 
![event_schema_and_pipeline](https://user-images.githubusercontent.com/16739072/216821057-9dfcf4a4-955e-40cb-a361-f9abcae369f1.png)

**Reply by flomickl** [reply]
An alternative stream visualization with multiple events with different ID's could be:
![pipeline_multiEvents](https://user-images.githubusercontent.com/16739072/216821119-4170a4d0-1650-48d5-ab68-a564b2700168.png)

**Reply by tenthe** [reply]
Hi @flomickl,
my suggestion would be to use more IIoT-like sensors in the documentation.
Maybe we can find a good example of a machine or production line.
What do you think about it?

**Reply by flomickl** [reply]
yes of course.
I used the car because it is based on the examples that already exist

As an example, would temperature or water level sensor measurements be more appropriate?
